scipy/optimize/_qap.py

```diff
@@ -0,0 +1,408 @@
+                is `init_weight` and :math:`K` is some random doubly
+                stochastic matrix.
+            init_n : int, positive (default = 1)
+                Number of random initializations of the starting
+                permutation matrix that the FAQ algorithm will undergo.
+            maxiter : int, positive (default = 30)
+                Integer specifying the max number of Franke-Wolfe iterations.
+                FAQ typically converges with modest number of iterations.
+            shuffle_input : bool (default = True)
+                To avoid artificially high or low matching due to inherent
+                sorting of input adjacency matrices, gives users the option
+                to shuffle the nodes of `cost_matrix`. Results are then
+                unshuffled so that returned `col_ind` matches the node order
+                of inputs.
+            eps : float (default = 0.05)
+                A positive, threshold stopping criteria such that Franke-
+                Wolfe continues to iterate while Frobenius norm of
+                :math:`(P_{i}-P_{i+1}) > eps`, where :math:`i` is the
+                iteration number.
+
+    Returns
+    -------
+    res : OptimizeResult
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+
+            col_ind : 1-D array
+                An array of column indices corresponding to the optimal
+                permutation (with the fixed seeds given) of the
+                nodes of `dist_matrix`, to best minimize the objective
+                function.
+            score : float
+                The optimal value of the objective function.
+            nit : int
+                The total number of Franke-Wolfe iterations performed during
+                optimization.
+
+    References
+    ----------
+    .. [1] J.T. Vogelstein, J.M. Conroy, V. Lyzinski, L.J. Podrazik,
+           S.G. Kratzer, E.T. Harley, D.E. Fishkind, R.J. Vogelstein, and
+           C.E. Priebe, "Fast approximate quadratic programming for graph
+           matching," PLOS one, vol. 10, no. 4, p. e0121002, 2015,
+           https://doi.org/10.1371/journal.pone.0121002
+
+    .. [2] D. Fishkind, S. Adali, H. Patsolic, L. Meng, D. Singh, V. Lyzinski,
+           C. Priebe, "Seeded graph matching", Pattern Recognit. 87 (2019):
+           203-215, https://doi.org/10.1016/j.patcog.2018.09.014
+
+    Examples
+    --------
+
+    >>> cost = np.array([[0, 80, 150, 170], [80, 0, 130, 100],
+    ...         [150, 130, 0, 120], [170, 100, 120, 0]])
+    >>> dist = np.array([[0, 5, 2, 7], [0, 0, 3, 8],
+    ...         [0, 0, 0, 3], [0, 0, 0, 0]])
+    >>> from scipy.optimize import quadratic_assignment
+    >>> res = quadratic_assignment(cost, dist)
+    >>> print(res)
+     col_ind: array([0, 3, 2, 1])
+         nit: 9
+       score: 3260
+
+    To demonstrate explicitly how the `score` value
+    :math:`f(P) = trace(A^T PBP^T )` is calculated, one may construct the
```

Comment:
Even if it doesn't matter, I'd stick with
```suggestion
    :math:`f(P) = trace(A PB^TP^T )` is calculated, one may construct the
```
for consistency.